src/irc/core/irc-expandos.c

```diff
@@ -76,6 +76,32 @@ static char *expando_userhost(SERVER_REC *server, void *item, int *free_ret)
 	return g_strconcat(username, "@", hostname, NULL);;
 }
 
+/* your hostname address (host) */
+static char *expando_hostname(SERVER_REC *server, void *item, int *free_ret)
+{
+	IRC_SERVER_REC *ircserver;
+	char hostname[100];
+	char **list;
+	char *hostname_split;
+
+	ircserver = IRC_SERVER(server);
+
+	/* prefer the _real_ /userhost reply */
+	if (ircserver != NULL && ircserver->userhost != NULL) {
+		list = g_strsplit(ircserver->userhost, "@", -1);
+		hostname_split = g_strdup(list[1]);
+		g_strfreev(list);
+		return hostname_split;
```

Comment:
`*free_ret` must also be set to `TRUE` in this case, given that you return some newly initialised memory from your call to `g_strdup()`.